src/test/java/teammates/test/cases/browsertests/StudentProfilePageUiTest.java

```diff
@@ -332,4 +333,26 @@ private StudentProfilePage getProfilePageForStudent(String studentId) {
         return loginAdminToPage(profileUrl, StudentProfilePage.class);
     }
 
+    @AfterClass
+    public void classTearDown() {
+        // The data associated with the test accounts have to be removed when the test ends because
+        // the test accounts are shared across tests. In particular, the courses of test student1 account is not the same as
+        // `StudentHomePageUiTest` so it has to be cleared before running that test.
+
+        // Test data is never cleared after test in order to save datastore costs as we already remove and restore the data
+        // bundle when the test runs. This raises the question of why we do not remove the data bundle when the class
+        // ends then we do not have to remove it from startup. The reason is that in the case a test fails to tear down
+        // (fail to remove data bundle), another test should have no reason to fail.
```

Comment:
A general note: code comments should explain the current state of the code. Someone reading this code need not know  that this line is _being added_ at some point; rather, the comment should explain why the line _is_ there as if it was there from the very beginning. The history of how the code came to be like that should be in commit messages and issue/PR discussions.

Reply:
@tshradheya I forgot to mention that we should also document this in `removeAndRestoreDataBundle`.

I guess why the explanation here is a little strange because this really belongs there but we need to link it from here too. Maybe it's fine to duplicate the details in both places or you can link it from `removeAndRestoreDataBundle`.